test/extended/cpu_partitioning/crio.go

```diff
@@ -0,0 +1,314 @@
+		return 0, fmt.Errorf("err parsing cpushare json annotation: %w", err)
+	}
+
+	value, ok := temp["cpushares"]
+	if !ok {
+		return 0, fmt.Errorf("err cpushare annotation was incorrect expected format {\"cpushares\": <int> }")
+	}
+
+	return value, nil
+}
+
+var _ = g.Describe("[sig-node][apigroup:config.openshift.io] CPU Partitioning node validation", func() {
+	defer g.GinkgoRecover()
+
+	var (
+		oc                      = exutil.NewCLIWithoutNamespace("cpu-partitioning").AsAdmin()
+		ctx                     = context.Background()
+		isClusterCPUPartitioned = false
+	)
+
+	g.BeforeEach(func() {
+		isClusterCPUPartitioned = getCpuPartitionedStatus(oc) == ocpv1.CPUPartitioningAllNodes
+	})
+
+	g.It("crio containers should have correct cpuset and cpushare set", func() {
+		nodes, err := oc.AdminKubeClient().CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+		o.Expect(err).ToNot(o.HaveOccurred(), "error listing cluster nodes")
+
+		nodeRoleCPUSet, err := getExpectedCPUSetConstraints(ctx, oc, isClusterCPUPartitioned)
+		o.Expect(err).ToNot(o.HaveOccurred(), "error getting node CPUSets")
+
+		g.By(fmt.Sprintf("using nodeRoleCPUSet %s", nodeRoleCPUSet))
+
+		for _, node := range nodes.Items {
+			// Collect the container data from the node
+			crioData, err := collectContainerInfo(oc.AdminConfig(), oc.AsAdmin().KubeFramework().ClientSet, node)
+			o.Expect(err).ToNot(o.HaveOccurred(), fmt.Sprintf("error getting crio container data from node %s", node.Name))
+
+			// When the cluster is CPU Partitioned there should always be pods with workload annotations. However,
+			// we do not enforce this check on non partitioned clusters since we don't expect nodes beyond masters to have pods
+			// that contain workload annotations. In the event that they do, we allow the logic to continue to validate
+			// CPUSets.
+			if isClusterCPUPartitioned {
+				o.Expect(crioData).ToNot(o.BeEmpty(), "error getting crio container data, container data should not be empty")
+			}
+
+			// CPU Core counts are numbers 0-n, we minus one since Status.Capacity.Cpu count is numbered conventionally.
+			// We generate a wide open CPUSet to use for comparison.
+			totalCoreCount := node.Status.Capacity.Cpu().Value() - 1
+			parsedFullNodeCPUSet, err := cpuset.Parse(fmt.Sprintf("0-%d", totalCoreCount))
+			o.Expect(err).ToNot(o.HaveOccurred(), fmt.Sprintf("error creating full node cpuset, check totalCoreCount of (%d)", totalCoreCount))
+
+			expectedCPUSet := ""
+
+			g.By(fmt.Sprintf("fetching crio data from node (%s)", node.Name))
+			if _, isMaster := node.Labels[nodeMasterLabel]; isMaster {
+				expectedCPUSet = nodeRoleCPUSet[nodeMasterLabel]
+			} else if _, isWorker := node.Labels[nodeWorkerLabel]; isWorker {
+				expectedCPUSet = nodeRoleCPUSet[nodeWorkerLabel]
+			}
+
+			g.By(fmt.Sprintf("checking for expected CPUSet %s", expectedCPUSet))
+
+			for _, containerInfo := range crioData {
+				o.Expect(containerInfo.ContainerCPUSet).To(
+					o.Equal(expectedCPUSet),
```

Comment:
do we want or need we use the canonical cpuset string representation? should we use `cpuset.Equals` to be safe and avoid flakes? Also because you seem to use cpuset objects (vs their string representation) right below

Reply:
That's a good call out, the idea here was since this version of the CPUSet doesn't get parsed down to crio and is a 1/1 match on what the user placed in `reserved` that we would fail if it's otherwise, but that's not robust if things change. Updating